src/Hydration.js

```diff
@@ -3,22 +3,39 @@ class Hydration {
     this.hydrationData = hydrationData;
   }
   calculateAverageOunces(id) {
-    let perDayUserHydration = this.hydrationData.filter((data) => id === data.userID);
-    return perDayUserHydration.reduce((sumSoFar, data) => {
-      return sumSoFar += data.numOunces;
-    }, 0) / perDayUserHydration.length;
+    let hydrationForThisUser = this.hydrationData.filter((hydrationInstance) => id === hydrationInstance.userID);
+    const averageOuncesUnrounded =
+      hydrationForThisUser.reduce((sumSoFar, dayHydration) => {
+        return (sumSoFar += dayHydration.numOunces);
+      }, 0) / hydrationForThisUser.length;
+    return parseFloat(averageOuncesUnrounded.toFixed(2)) || undefined;
   }
+
   calculateDailyOunces(id, date) {
-    let findOuncesByDate = this.hydrationData.find((data) => id === data.userID && date === data.date);
-    return findOuncesByDate.numOunces;
+    const findOuncesByDate = this.hydrationData.find((data) => id === data.userID && date === data.date);
+    if (findOuncesByDate) {
+      return findOuncesByDate.numOunces;
+    }
+  }
+
+  //New function for the week of ounces
+  getWeekOfOunces(id, endDate) {
+    let hydrationForThisUser = this.hydrationData.filter((hydrationInstance) => id === hydrationInstance.userID);
+    const firstIndex = hydrationForThisUser.findIndex((day) => day.date === endDate);
+    const pastWeek = hydrationForThisUser.slice(firstIndex - 6, firstIndex + 1);
+    return pastWeek.map((day) => {
+      return { date: day.date, amount: day.numOunces, unit: 'oz' };
```

Comment:
I like this display of information much more, that way it's not an string on information but an object we can access info from.